### DISCUSSION TITLE
Installation instructions for WSL do not work

### DISCUSSION THREAD

**Discussion by b-hayes** [bridgetownrb/bridgetown#266] [Q&A]
From https://www.bridgetownrb.com/docs/installation/windows

After installing node and yarn the gem command fails saying I need ruby.
I install ruby with the suggested command `sudo apt install ruby` then continue.
I then had to run gem with sudo as it said I didn't have permission for the gems folder.
After the command `sudo gem install bridgetown -N` seemed fine but then I got stuck with.
```
bridgetown: command not found
```
IM stuck now, how do I get the bridgetown command?

**Comment by jaredcwhite** [comment]
Sorry to hear about that @b-hayes. Unfortunately I'm not in a position to test this out (not a Windows user). I do know installing gems via `sudo` is highly discouraged. I'd see if you can uninstall all that and then use something like `rbenv` to install Ruby/Bundler instead.

**Reply by b-hayes** [reply]
I had to use `sudo` otherwise Id get `You don't have write permissions for the /var/lib/gems/2.7.0 directory.`  I dont know what `rbenv` is.
```
Command 'rbenv' not found, but can be installed with:

sudo apt install rbenv
```
Will apt install the correct command?

Should I just take ownership of gems folder?

**Reply by jaredcwhite** [reply]
In the GoRails installation guide we recommend for Windows/WSL users, it talks about installing Ruby via rbenv. If you don't have write permissions to your gems folder, it's not going to work well. Check out: https://gorails.com/setup/windows/10#ruby-rbenv

**Reply by jaredcwhite** [reply]
For reference, I *only* ever use Ruby through a version manager like rbenv or RVM. It's nothing but pain otherwise!

**Reply by b-hayes** [reply]
Hey, thanks for that I already had ruby installed assuming it was fine without a version manager.
The installation from https://gorails.com/setup/windows/10#ruby-rbenv also had problems.
I recommend adding a new line before adding to `.bashrc` as without it appended to the last command and essentially broke everything.

Was running into `ruby -vrbenv: no such command install'` managed to recover from that one still trying to install it 😫.

Now im getting this:
```
configure: error: in `/tmp/ruby-build.20210412110034.23842.ulpFey/ruby-2.7.2':
configure: error: no acceptable C compiler found in $PATH
See `config.log' for more details
```

So many dependencies to resolve lol. Could setup an apache web server before I get this working.

**Reply by b-hayes** [reply]
Googled and then used `sudo apt install gcc` and progressed to this error
`BUILD FAILED (Ubuntu 20.04 using ruby-build 20210405-2-g693a28e)`
Pretty close to giving up on using bridgetown.

**Reply by b-hayes** [reply]
one step closer,
checked the log file and said it had no make command, I did
 `sudo apt install make` progressed to this error (from the build log)
 
 ```
 installing default gems from lib:   /home/brad/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0 (build_info, cache, doc, extensions, gems, specifications)
                                    benchmark 0.1.0
/tmp/ruby-build.20210412111653.11237.F0o5wp/ruby-2.7.2/lib/rubygems/core_ext/kernel_require.rb:92:in `require': cannot load such file -- openssl (LoadError)
	from /tmp/ruby-build.20210412111653.11237.F0o5wp/ruby-2.7.2/lib/rubygems/core_ext/kernel_require.rb:92:in `require'
	from /tmp/ruby-build.20210412111653.11237.F0o5wp/ruby-2.7.2/lib/rubygems/specification.rb:2430:in `to_ruby'
	from ./tool/rbinstall.rb:846:in `block (2 levels) in install_default_gem'
	from ./tool/rbinstall.rb:279:in `open_for_install'
	from ./tool/rbinstall.rb:845:in `block in install_default_gem'
	from ./tool/rbinstall.rb:835:in `each'
	from ./tool/rbinstall.rb:835:in `install_default_gem'
	from ./tool/rbinstall.rb:799:in `block in <main>'
	from ./tool/rbinstall.rb:950:in `block in <main>'
	from ./tool/rbinstall.rb:947:in `each'
	from ./tool/rbinstall.rb:947:in `<main>'
make: *** [uncommon.mk:373: do-install-all] Error 1
 ```

Not sure where to go from there?

**Reply by b-hayes** [reply]
Got it!!!

Reran the dependency commands above closed and reopened my terminal and then got ruby running!

Think I needed gcc and make installed before the other dependencies maybe for it to work?

**Reply by jaredcwhite** [reply]
Glad you finally got it working!! 😅

**Reply by b-hayes** [reply]
Yeah same. Thanks for the help.
